src/platform/file_manager_cache.c

```diff
@@ -168,4 +169,11 @@ int platform_file_manager_cache_file_has_extension(const file_info *f, const cha
     return platform_file_manager_compare_filename(f->extension, extension) == 0;
 }
 
+bool platform_file_manager_cache_has_stat(void)
+{
+    if (stat_status == STAT_UNTESTED)
+        return false;
+    return stat_status == STAT_WORKS;
```

Comment:
You don't need the first check as `STAT_WORKS` is already different than `STAT_UNTESTED`.